client/src/pages/sponsorsGuests.tsx

```diff
@@ -0,0 +1,86 @@
+import { Montserrat } from "next/font/google";
+import React, { useEffect, useState } from "react";
+
+import { GuestSection, SponsorSection } from "@/components/SponsorsGuests";
+
+type Partner = {
+  name: string;
+  logo: string;
+  link: string;
+};
+
+// Type definition for guest data
+type Guest = {
+  name: string;
+  description: string;
+  link: string;
+  photo?: string;
+};
+
+const montserrat = Montserrat({
+  subsets: ["latin"],
+  weight: ["400", "600", "700"],
+});
+
+const SponsorsAndGuest = () => {
+  const [partners, setPartners] = useState<Partner[]>([]); // Holds sponsor data
+  const [guests, setGuests] = useState<Guest[]>([]); // Holds guest data
+  const [loading, setLoading] = useState(true); // Controls loading state
+
+  useEffect(() => {
+    // Define an async function to fetch data
+    const fetchData = async () => {
+      try {
+        const [partnerRes, guestRes] = await Promise.all([
+          fetch("http://localhost:8000/api/sponsor/"),
+          fetch("http://localhost:8000/api/sponsor/guests/"),
```

Comment:
Not a big deal right now but environment variables tend to be preferred.

Make sure to define `NEXT_PUBLIC_API_BASE_URL` in `.env.local`

```
const BASE_URL = process.env.NEXT_PUBLIC_API_BASE_URL || "http://localhost:8000";

const [partnerRes, guestRes] = await Promise.all([
  fetch(`${BASE_URL}/api/sponsor/`),
  fetch(`${BASE_URL}/api/sponsor/guests/`),
]);
```